frontend/src/domain/entities/vessel/track/index.test.ts

```diff
@@ -0,0 +1,276 @@
+import { describe, expect, it } from '@jest/globals'
+
+import { getFeaturesFromPositions, getTrackType } from './index'
+
+import type { VesselLineFeature, VesselPointFeature, VesselPosition } from '../types'
+
+describe('vessel/track', () => {
+  it('getFeaturesFromPositions Should return one feature point When one position is given', async () => {
+    // Given
+    const positions = [
+      {
+        course: 356,
+        dateTime: '2022-10-27T16:10:20.4564+02:00',
+        destination: 'DE',
+        externalReferenceNumber: 'SE457432',
+        flagState: 'FR',
+        from: 'FR',
+        internalReferenceNumber: 'ABC000898396',
+        ircs: 'ZBRI',
+        isFishing: null,
+        isManual: null,
+        latitude: 46.386,
+        longitude: -3.328,
+        mmsi: null,
+        positionType: 'VMS',
+        speed: 2.8,
+        tripNumber: null,
+        vesselName: 'CE DEVANT ÉLEVER'
+      }
+    ]
+    const vesselId = 'VESSEL_ID'
+
+    // When
+    const features = getFeaturesFromPositions(positions, vesselId)
+
+    // Then
+    expect(features).toHaveLength(1)
+
+    const feature = features[0] as VesselPointFeature
+    expect(feature!.course).toEqual(356)
```

Comment:
You can use `.toMatchObject()` there and the following ones to make it easier ;).

And you can use, when needed, other expects inside like that for example:

```ts
expect(myObject).toMatchObject({
  a: 'a string',
  b: expect.stringMatching('a match'),
  c: expect.any(Number)
})
```